server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/MailMatcher.java

```diff
@@ -48,36 +48,38 @@
 public interface MailMatcher {
 
     interface HeaderExtractor extends ThrowingFunction<Mail, Stream<String>> {
```

Comment:
what is the value to keep these classes inner?

Reply:
Well, we can very well make them TLD classes, should we?

I imagine because we considered the amount of code written did not deserve it.

Reply:
That's just a question, I just want to know the rationals

Reply:
Anyway I extracted it, it seems also more clear to me...

TooMuchNestedClassException & TooMuchRegistryInTheSameFileException